charts/fybrik/notebook-test-readflow.tls.values.yaml

```diff
@@ -90,22 +90,16 @@ opaServer:
 
   # Bootstrap policies to load upon startup
   bootstrapPolicies:
-    notebookSamplePolicy: |-
-      package dataapi.authz
-
-      rule[{}] {
-        description := "allow read datasets with no tags"
-        input.action.actionType == "read"
-        not input.resource.metadata.tags
-      }
-
-      rule[{"action": {"name":"RedactAction", "columns": column_names}, "policy": description}] {
-        description := "While reading redact columns tagged as PII in datasets tagged with finance = true"
-        input.action.actionType == "read"
-        input.resource.metadata.tags.finance
-        column_names := [input.resource.metadata.columns[i].name | input.resource.metadata.columns[i].tags.PII]
-      }
-
+    notebookSamplePolicy: |+
+        package dataapi.authz
+
+        rule[{"action": {"name":"RedactAction", "columns": column_names}, "policy": description}] {
+           description := "Redact columns tagged as PII.Sensitive in datasets tagged with Purpose.finance = true"
+           input.action.actionType == "read"
+           input.resource.metadata.tags["Purpose.finance"]
+           column_names := [input.resource.metadata.columns[i].name | input.resource.metadata.columns[i].tags["PII.Sensitive"]]
+           count(column_names) > 0
```

Comment:
> Why do we check `count(column_names) > 0`? Or this test is with allow-by-default?

The redaction rule should not be triggered if there is no data to redact. If no rules are triggered, the default policy will be applied - allow or not.

Reply:
if the default policy is `deny by default` then the `count(column_names) > 0` condition is needed to deny assets with "Purpose.finance" tag and that do not contain "PII.Sensitive" tags. because otherwise an action object with empty columns is returned while with `count(column_names) > 0` the rule will not be set and if no other rule exists then the request will be denied.
This is a link to opa playground showing this case https://play.openpolicyagent.org/p/MUAzu72S2n
When the default policy is `allow by default` the `count(column_names) > 0` condition is redundant because with this condition and without it the request will not be denied: with `count(column_names) > 0` allow by default will be applied given that no other rule exists and without `count(column_names) > 0` action is returned with empty columns.

Reply:
This condition is not redundant. Suppose you have no sensitive data and you decide not to deploy a redaction module. What will happen is that a Redact action will be returned (with empty columns), Fybrik will fail to find a module that does redaction, thus, failing to create a data plane. The request will be stuck in an error state waiting for a module to be deployed...

Reply:
@shlomitk1 OK thanks! I will add this explanation to the condition comment per @tomersolomon1 request.

Reply:
I suggest adding the following comment:
```
If column_names is empty this rule will not be triggered (and the default rule will be triggered instead). Note however that currently in the test there are columns tagged as sensitive, and furthermore the default is set to allow - so practically this behaves the same with and without this condition. However, we check the length of column_names as an example of good policy writing. Since currently even when column_names is empty a reduction module will be deployed, which will do nothing, and is therefore a waste of resources.
```

Reply:
@tomersolomon1 Thanks.  I ended up adding a short explanation in a comment.